core/src/components/cat-tooltip/cat-tooltip.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, h, Host, Prop} from '@stencil/core';
+import {autoUpdate, computePosition, flip, offset, Placement} from "@floating-ui/dom";
+
+@Component({
+  tag: 'cat-tooltip',
+  styleUrl: 'cat-tooltip.scss',
+  shadow: true,
+})
+export class CatTooltip {
+  private tooltip?: HTMLElement;
+  private trigger?: Element;
+  private showTimeout: NodeJS.Timeout;
+  private hideTimeout: NodeJS.Timeout;
+
+  /**
+   * The content of the tooltip
+   */
+  @Prop() content: string;
+
+  /**
+   * Specifies that the tooltip should be disabled. A disabled tooltip is unusable,
+   * and invisible. Corresponds with the native HTML disabled attribute.
+   */
+  @Prop() disabled = false;
+
+  /**
+   * The placement of the tooltip.
+   */
+  @Prop() placement: Placement = 'top';
+
+  /**
+   * The delay time for showing tooltip in ms
+   */
+  @Prop() showDelay = 500;
+
+  /**
+   * The delay time for hiding tooltip in ms
+   */
+  @Prop() hideDelay = 500;
+
+  componentDidLoad(): void {
+    if (this.trigger && this.tooltip) {
+      autoUpdate(this.trigger, this.tooltip, () => this.update());
+    }
+    this.trigger?.addEventListener('mouseenter', this.showListener.bind(this));
+
+    this.trigger?.addEventListener('mouseleave', this.hideListener.bind(this));
+  }
+
+  disconnectedCallback(): void {
+    this.trigger?.removeEventListener('mouseenter', this.showListener.bind(this));
+    this.trigger?.removeEventListener('mouseleave', this.hideListener.bind(this));
+  }
+
+  render() {
+    return (
+      <Host>
+        <div class="tooltip-content" ref={el => (this.trigger = el)}>
```

Comment:
Shouldnt this rather be called tooltip-trigger? And it should also have display: contents, no?